aea/aea_builder.py

```diff
@@ -1385,6 +1388,27 @@ def _load_and_add_components(
                 component = load_component_from_config(configuration, **kwargs)
                 resources.add_component(component)
 
+    def _populate_contract_registry(self):
```

Comment:
this is called from `build`. Basically, whenever we build an agent, we also register contracts in the global contract registry.

This might be a problem because we also register the contract configuration to initialize the contract, and since it is a global state it can be accessed by other agents in the same process.

Reply:
Alternative: attach the registry to the `LedgerApiConnection` instance.